feat: DIA-2149: Only sync new annotations in export sync cronjob

matt-bernstein opened the pull request:
Add new export storage sync option to only look at net-new annotations, to be used during nightly syncs to avoid redundant syncing of existing annotations
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

matt-bernstein committed:
feat: DIA-2149: [BE] Only sync new annotations in export sync cronjob

matt-bernstein committed:
add option to sync only new annots to export storage

matt-bernstein commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14380297449)
> Successfully merged:  1 file changed, 57 insertions(+), 3 deletions(-)

matt-bernstein committed:
Merge branch 'develop' into 'fb-dia-2149'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14380297449

pakelley commented on label_studio/io_storages/base_models.py:
```diff
@@ -496,6 +496,12 @@ def export_sync_background(storage_class, storage_id, **kwargs):
 
 
+@job('low', timeout=settings.RQ_LONG_JOB_TIMEOUT)
+def export_sync_background_only_new(storage_class, storage_id, **kwargs):
```
nit: mind naming this `export_sync_only_new_background`?

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-dia-2149

matt-bernstein committed:
rename

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-dia-2149

matt-bernstein commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14389509283)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14389509283
